lapis-e2e/test/details.spec.ts

```diff
@@ -220,4 +253,128 @@ key_1002052
 
     expect(result).to.have.nested.property('data[0].division', null);
   });
+
+  it('variantQuery and advancedQuery should be the same for sequence and regex intersections and unions', async () => {
+    const sequenceQueries = [
+      '300G & !400- & (S:123T | S:234A)',
+      '[3-of: 123A, 234T, S:345-, ORF1a:456K, ORF7A:100-]',
+      '[exactly-2-of: 123A & 234T, !234T, S:345- | S:346-, [2-of: 222T, 333G, 444A, 555C]]',
+      'MAYBE(123W)',
+    ];
+    const regexQueries = ['region\d', 'Basel-(Stadt|Land)', '[^a-c]', 'Basel.*', '^Z.*rich$', 'region{1,2}'];
+    for (var sequenceQuery of sequenceQueries) {
+      for (var regexQuery of regexQueries) {
+        const urlRegex = new URLSearchParams({
+          'fields': 'primaryKey',
+          'division.regex': regexQuery,
+          'orderBy': 'primaryKey',
+          'dataFormat': 'csv',
+        });
+
+        const urlVariant = new URLSearchParams({
+          fields: 'primaryKey',
+          variantQuery: sequenceQuery,
+          orderBy: 'primaryKey',
+          dataFormat: 'csv',
+        });
+
+        const resultRegex = await fetch(basePath + '/sample/details?' + urlRegex.toString());
+        const resultVariant = await fetch(basePath + '/sample/details?' + urlVariant.toString());
+
+        expect(resultRegex.status).to.be.equal(200);
+        expect(resultVariant.status).to.be.equal(200);
+
+        const resultRegexText = await resultRegex.text();
+        const resultVariantText = await resultVariant.text();
+        const setRegex = new Set(resultRegexText.split('\n').filter(Boolean));
+        const setVariant = new Set(resultVariantText.split('\n').filter(Boolean));
```

Comment:
Why did you request the data as CSV when you decompose it here into a set again? LAPIS can give us a list in a JSON, then you don't need to "parse" CSV in this test.